### PR TITLE
fixes for new check errors in R >= 4.4.0

### PR DESCRIPTION
**Pull request by infotroph**
## Description

Ran `make check` under R 4.4.1 and worked through the new complaints, either fixing issues outright (where it was easy) or updating the log (where it will take more work to resolve the message). 

Predictably, the project ballooned and touched a lot of files, so this PR focuses on the packages in `base/` plus the first few `modules/`. I'll follow up with at least two more for `modules/` and `models/`.

## Motivation and Context

R >= 4.4 introduced some nice refinements to the wording and content of R's package check messages. I'm glad to see these improved to benefit the whole R community, but they create some work for us in the short term by re-surfacing some of the messages we've been ignoring in Rcheck_reference.log.

"Wait, Chris", you say. "Don't we run checks against R-devel every week? Why didn't we notice these changes before?" Well, yes, but those tests build off `rocker/tidyverse:devel`, whose updates turn out to be [stalled because of an error in the automated build](https://github.com/rocker-org/rocker-versioned2/issues/822). So our weekly checks have been rechecking against the same R-devel daily snapshot since April, (coincidentally?) just before the release of R 4.4.0.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: base/db/tests/Rcheck_reference.log [hunk 1]
```diff
@@ -62,7 +62,8 @@ The Date field is over a month old.
 * checking package directory ... OK
 * checking for future file timestamps ... OK
 * checking ‘build’ directory ... OK
-* checking DESCRIPTION meta-information ... OK
+* checking DESCRIPTION meta-information ... NOTE
+License stub is invalid DCF.
```
Comment (infotroph) [inline base/db/tests/Rcheck_reference.log:66]: This note is newly added in R 4.4. the best way to fix it is to finish relicensing each package as BSD.

File: modules/allometry/R/AllomAve.R [hunk 2]
```diff
@@ -7,16 +7,27 @@
 # http://opensource.ncsa.illinois.edu/license.html
 #-------------------------------------------------------------------------------
 
-#' @title AllomAve
-#' @name  AllomAve
-#' @aliases AllomAve
+#' AllomAve
+#'
+#' Allometery wrapper function that handles loading and subsetting the data,
```
Comment (infotroph) [inline modules/allometry/R/AllomAve.R]:
```suggestion
#' Allometry wrapper function that handles loading and subsetting the data,
```

File: modules/allometry/man/AllomAve.Rd [hunk 3]
```diff
@@ -47,15 +48,16 @@ If the latter, the names within the list are used to identify PFTs
 nested list of parameter summary statistics
 }
 \description{
-allometery wrapper function that handles loading and subsetting the data,
+Allometery wrapper function that handles loading and subsetting the data,
```
Comment (infotroph) [inline modules/allometry/man/AllomAve.Rd]:
```suggestion
Allometry wrapper function that handles loading and subsetting the data,
```

File: modules/assim.sequential/R/Analysis_sda.R [hunk 4]
```diff
@@ -98,6 +98,7 @@ EnKF<-function(settings, Forecast, Observed, H, extraArg=NULL, ...){
 ##' @param settings  pecan standard settings list.  
 ##' @param Forecast A list containing the forecasts variables including Q (process variance) and X (a dataframe of forecast state variables for different ensemble)
 ##' @param Observed A list containing the observed variables including R (cov of observed state variables) and Y (vector of estimated mean of observed state variables)
+##' @param H not used
```
Comment (infotroph) [inline modules/assim.sequential/R/Analysis_sda.R:101]: @DongchenZ can you confirm I got this right?

File: modules/assim.sequential/R/Multi_Site_Constructors.R [hunk 5]
```diff
@@ -6,6 +6,9 @@
 ##' @param var.names vector names of state variable names.
 ##' @param X a matrix of state variables. In this matrix rows represent ensembles, while columns show the variables for different sites.
 ##' @param localization.FUN This is the function that performs the localization of the Pf matrix and it returns a localized matrix with the same dimensions.
+##' @param t not used
+##' @param blocked.dis passed to `localization.FUN`
+##' @param ... passed to `localization.FUN`
```
Comment (infotroph) [inline modules/assim.sequential/R/Multi_Site_Constructors.R:11]: @DongchenZ Come to think of it, can you please check all the places in assim.sequential where I added missing `@param` lines? You know this package better than me.

File: modules/assim.sequential/R/load_data_paleon_sda.R [hunk 6]
```diff
@@ -294,7 +294,7 @@ load_data_paleon_sda <- function(settings){
     
     ### Error Message for no data product
     if(format_id[[i]] != '1000000040' & format_id[[i]] != '1000000058'){
-      PEcAn.logger::logger.severe('ERROR: This data format has not been added to this function (ツ)_/¯ ')
```
Comment (infotroph) [inline modules/assim.sequential/R/load_data_paleon_sda.R:297]: R hates fun and complains about the non-ASCII characters in this string (ツ)_/¯

File: modules/assim.sequential/R/Analysis_sda.R (hunk 4 above)
Comment (DongchenZ) [reply modules/assim.sequential/R/Analysis_sda.R:101]: yes, it's not used in this function.

File: modules/assim.sequential/R/Multi_Site_Constructors.R (hunk 5 above)
Comment (DongchenZ) [reply modules/assim.sequential/R/Multi_Site_Constructors.R:11]: They look good to me.

Comment (infotroph) [reply modules/assim.sequential/R/Multi_Site_Constructors.R:11]: Thank you for checking!

File: modules/allometry/R/allom.predict.R [hunk 7]
```diff
@@ -7,27 +7,29 @@
...
 #' }
 #' @param dbh Diameter at Breast Height (cm)
-#' @param pft Plant Functional Type. Needs to match the name used in AllomAve. Can be NULL if only one PFT/species exists, otherwise needs to the same length as dbh
+#' @param pft Plant Functional Type. Needs to match the name used in AllomAve.
+#'  Can be NULL if only one PFT/species exists, otherwise needs to the same length as dbh
 #' @param component Which component to predict. Can be NULL if only one component was analysed in AllomAve.
 #' @param n Number of Monte Carlo samples. Defaults to the same number as in the MCMC object
 #' @param use  c('Bg','mu','best')
 #' @param interval c('none','confidence','prediction') default is prediction
+#' @param single.tree logical: Is this a DBH time series from one indidual tree?
```
Comment (meetagrawal09) [inline modules/allometry/R/allom.predict.R]: NIT : indidual -> individual.

Comment (infotroph) [reply modules/allometry/R/allom.predict.R]: fixed in 2481f31c7

**Comment by infotroph** [comment]
@meetagrawal09 and @DongchenZ, have you both finished your reviews of this? Have you been through it enough to give an overall 👍 / 👎 ?

**Review by meetagrawal09 — APPROVED** [review]

**Pull request merged**